tests/iterator-tests.lisp

```diff
@@ -2,6 +2,8 @@
 
 (define-test iter-list-iter-collect-list-id ()
   (is (== (the (List Integer) (make-list 0 1 2 3))
+          ;; FIXME: changing this to `iter:collect! (iter:into-iter (make-list 0 1 2 3))' breaks type
```

Comment:
ISTM that we should sort this out before merging

Reply:
agreed